app/forms/validators.py

```diff
@@ -49,15 +54,15 @@ def __call__(
         field: Union[DecimalFieldWithSeparator, IntegerFieldWithSeparator],
     ) -> None:
         try:
-            Decimal(
-                field.raw_data[0].replace(
-                    numbers.get_group_symbol(flask_babel.get_locale()), ""
-                )
-            )
+            Decimal(sanitise_number(number=field.raw_data[0]))
         except (ValueError, TypeError, InvalidOperation, AttributeError) as exc:
             raise validators.StopValidation(self.message) from exc
 
-        if "e" in field.raw_data[0].lower():
+        if "e" in field.raw_data[0].lower() or math.isnan(
+            Decimal(
+                sanitise_number(number=field.raw_data[0])
+            )  # number is sanitised to guard against inputs like `,NaN_` etc
+        ):
```

Comment:
How come this changed?. What broke it?.

Reply:
Test percentage schema has integer with zero decimals input that, when it arrives at `NumberCheck` validator, has already `field.data` value set as `None`, so it was passing validation when you entered `"NaN"`. We have to use `raw_data` here?